File: Greedy/Cpp/2217.cpp
// 로프
#include <algorithm>
#include <iostream>
using namespace std;
int main() {
    int N;
    cin >> N;
    int *lope = new int[N];
    for (int i = 0; i < N; i++) {
        cin >> lope[i];
    }
    sort(lope, lope + N);
    int max = -1;
    int weight;
    int count = 1;
    for (int i = N - 1; i >= 0; i--) {
        weight = lope[i] * count;
        if (max < weight) max = weight;
        count++;
    }
    cout << max;
    delete[] lope;
}

File: Greedy/Python/2217.py
# 로프
N = int(input())
lope = []
for i in range(N):
    lope.append(int(input()))
lope.sort(reverse=True)
maxx = -1
for i in range(N):
    weight = lope[i] * (i+1)
    if maxx < weight:
        maxx = weight
print(maxx)
